docs/integrations/sources/s3.md

```diff
@@ -152,66 +172,105 @@ Please note, the S3 Source connector used to infer schemas from all the availabl
+- **Allow newlines in values**: Also known as _multiline_, this option addresses situations where newline characters occur within text data. Typically, newline characters signify the end of a row in a CSV, but when this option is set to True, parsing correctly handles newlines within values. Set to False by default.
+- **Additional Reader Options**: This allows for editing the less commonly used CSV [ConvertOptions](https://arrow.apache.org/docs/python/generated/pyarrow.csv.ConvertOptions.html#pyarrow.csv.ConvertOptions). The value must be a valid JSON string, e.g.:
+
+  ```
+  {
+    "timestamp_parsers": [
+    "%m/%d/%Y %H:%M", "%Y/%m/%d %H:%M"
+    ], 
+    "strings_can_be_null": true, 
+    "null_values": [
+      "NA", "NULL"
+    ]
+  }
   ```
 
-- `advanced_options` : This allows for editing the less commonly required CSV [ReadOptions](https://arrow.apache.org/docs/python/generated/pyarrow.csv.ReadOptions.html#pyarrow.csv.ReadOptions). The value must be a valid JSON string. One use case for this is when your CSV has no header, or you want to use custom column names, you can specify `column_names` using this option.
+- **Advanced Options**: This allows for editing the less commonly used CSV [ReadOptions](https://arrow.apache.org/docs/python/generated/pyarrow.csv.ReadOptions.html#pyarrow.csv.ReadOptions). The value must be a valid JSON string. One use case for this is when your CSV has no header, or if you want to use custom column names. You can specify `column_names` using this option. For example:
 
-  ```test
-  {"column_names": ["column1", "column2", "column3"]}
   ```
+  {
+    "column_names": [
+      "column1", "column2", "column3"
+    ]
+  }
+  ```
+
+- **Block Size**: This is the number of bytes to process in memory at a time while reading files. The default value of `10000` is usually suitable, but if your table is particularly wide \(lots of columns / data in fields is large\), increasing this might help avoid schema detection failures.
+
+:::caution
+Be cautious when raising this value too high, as it may result in Out Of Memory issues due to increased memory usage.
+:::
 
 ### Parquet
 
-Apache Parquet file is a column-oriented data storage format of the Apache Hadoop ecosystem. It provides efficient data compression and encoding schemes with enhanced performance to handle complex data in bulk. For now, the solution involves iterating through individual files at the abstract level thus partitioned parquet datasets are unsupported. The following settings are available:
+Apache Parquet is a column-oriented data storage format of the Apache Hadoop ecosystem. It provides efficient data compression and encoding schemes with enhanced performance to handle complex data in bulk. For now, the solution involves iterating through individual files at the abstract level. Therefore, partitioned parquet datasets are unsupported. The following settings are available:
```

Comment:
This sentence is hard to understand as a user so I just suggest removing it and saying partitioned datasets are unsupported
```suggestion
Apache Parquet is a column-oriented data storage format of the Apache Hadoop ecosystem. It provides efficient data compression and encoding schemes with enhanced performance to handle complex data in bulk. At the moment, partitioned parquet datasets are unsupported. The following settings are available:
```